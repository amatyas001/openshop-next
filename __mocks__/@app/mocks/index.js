export const mockProduct = (repeat = 1) => {
  let index = 0;
  let items = [];
  do {
    items.push({
      id: `mock_id_${index}`,
      amount: 1,
      color: `mock_color_${index}`,
      description: `mock_description_${index}`,
      gender: `mock_gender_${index}`,
      img: `mock_img_${index}`,
      name: `mock_name_${index}`,
      price: 1,
      review: `mock_review_${index}`,
      sizes: [`mock_size_0_${index}`, `mock_size_1_${index}`],
      starrating: 1,
    });
  } while (--repeat !== 0);
  return items;
};
