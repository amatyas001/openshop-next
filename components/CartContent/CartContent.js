import Router from 'next/router';
import { useSelector, useDispatch } from 'react-redux';
import { getAmount } from '../../redux/actions';
import { Heading } from '@chakra-ui/core';
import { CartItem } from '../index';

export async function getStaticProps() {
  return {
    props: { initialReduxState: { cart: [] } },
  };
}

// Displays all items which added to the cart
export const CartContent = () => {
  const { cart, amount } = useSelector((state) => state);
  const dispatch = useDispatch();

  // prefetching item info paths
  React.useEffect(() => {
    const prefetchRoutes = (url) => {
      url.match(/^\/item\/[\w]*$/gi) && Router.prefetch('/items/[id]', url);
    };
    Router.events.on('beforeHistoryChange', prefetchRoutes);
    return () => {
      Router.events.off('beforeHistoryChange', prefetchRoutes);
    };
  }, []);

  // update amount according to changes
  React.useEffect(() => {
    dispatch(getAmount());
  }, [cart]);

  return (
    <section id='cart-holder' role='list'>
      {/* items */}
      {cart.map((i) => (
        <CartItem key={i.id} item={i} />
      ))}

      <footer>
        {/* divider */}
        <Heading
          as='h3'
          id='total-summary'
          textAlign='right'
          fontSize='1.5rem'
          fontWeight='bold'
          color='purple.500'
          pt='10px'
          my='0'
          borderTop='1px'
        >
          TOTAL PRICE
        </Heading>

        {/* price */}
        <Heading
          as={'strong'}
          id='total-price'
          d='block'
          textAlign='right'
          aria-describedby='total-summary'
          my='0'
        >
          {amount}&nbsp;$
        </Heading>
      </footer>
    </section>
  );
};
