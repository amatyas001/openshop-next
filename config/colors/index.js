export * from './_button';
export * from './_panel';
export * from './_general';
