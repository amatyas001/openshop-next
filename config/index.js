export * from './colors';
export * from './menus/footer';
export * from './menus/main';
export * from './fields/payment';
export * from './state/initial';
