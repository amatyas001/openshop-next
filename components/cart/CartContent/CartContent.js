import PropTypes from 'prop-types';
import { useSelector } from 'react-redux';
import { Box } from '@chakra-ui/core';
import { CartItem } from '@app/components';

/**
 * Wrapper for listing products added to the cart
 *
 * @see https://amatyas001.github.io/openshop-next/#cartpanel
 */
export const CartContent = (props) => {
  const { icons, details } = props;
  const { cart } = useSelector((state) => state);
  return (
    <Box as='ul' role='list' width='90%' {...props}>
      {cart.map((item) => (
        <CartItem
          key={item.id}
          role='listitem'
          as='li'
          item={item}
          icons={icons}
          details={details}
        />
      ))}
    </Box>
  );
};

CartContent.defaultProps = {
  icons: false,
  details: false,
};

CartContent.propTypes = {
  /**
   * Show controls
   */
  icons: PropTypes.bool,

  /**
   * Show description
   */
  details: PropTypes.bool,
};
