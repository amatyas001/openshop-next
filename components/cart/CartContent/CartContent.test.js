import { create, act } from 'react-test-renderer';
import { CartContent } from './CartContent';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { INITIAL_STATE } from '@app/config/';
import { mockProduct } from '@app/__mocks__/@app/mocks';

const mockStore = configureStore([]);

describe('<CartContent />', () => {
  let tree, store;

  it('should render without props', () => {
    act(() => {
      tree = create(
        <Provider store={mockStore(INITIAL_STATE)}>
          <CartContent />
        </Provider>
      );
    });
    expect(tree.toJSON()).toMatchSnapshot();
  });

  describe('on non empty cart', () => {
    beforeAll(() => {
      store = mockStore({
        ...INITIAL_STATE,
        cart: [{ ...mockProduct(1)[0], buy: { amount: 1 } }],
      });

      act(() => {
        tree = create(
          <Provider store={store}>
            <CartContent />
          </Provider>
        );
      });
    });

    it('should render cart items', () => {
      expect(tree.root.findByType('ul').props.children).toHaveLength(1);
    });
  });
});
