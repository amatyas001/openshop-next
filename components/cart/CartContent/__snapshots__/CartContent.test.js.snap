// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CartContent /> should render without props 1`] = `
<ul
  className="css-7822g1"
  role="list"
/>
`;
